src/meta/src/manager/catalog/fragment.rs

```diff
@@ -1120,9 +1120,20 @@ impl FragmentManager {
             .filter(|t| t.fragment_ids().any(|f| reschedules.contains_key(&f)))
             .map(|t| t.table_id())
             .collect_vec();
+
+        let fragment_id_to_table_id: HashMap<_, _> = map
+            .iter()
+            .flat_map(|(table_id, table)| {
+                table
+                    .fragment_ids()
+                    .map(|fragment_id| (fragment_id, *table_id))
+            })
+            .collect();
+
```

Comment:
We should probably add a tracking issue for this, to make it part of the fragment manager's fields.

Because here, for each `alter` we need to rebuild this `fragment_id_to_table_id` map. We should just update it incrementally.